### ISSUE TITLE
Provide debug information about what line errors in the linter are coming from

### ISSUE THREAD

**Issue by nlarusstone** [r-lib/lintr#768]
Similar to #461 I'm seeing the following error in my output:

```
..Error in vapply(eq_assign_locs, next_with_parent, pc = pc, integer(1)) : 
  values must be length 1,
 but FUN(X[[2]]) result is length 0
 ```

but I can't figure out where the error is. It would be very helpful to have a debug statement that points out what line this error occurred on

**Comment by MichaelChirico** [comment]
could you confirm your version info? please be sure to try your code on the current dev version

**Comment by nlarusstone** [comment]
```
> packageVersion("lintr")
[1] ‘2.0.1’
```
Are you saying this error message has been improved since that version?

(btw, I have solved my specific error, but just wanted to open this issue because it was quite difficult to track down and I feel like `lintr` could be quite helpful here)

**Comment by MichaelChirico** [comment]
> Are you saying this error message has been improved since that version?

Not quite, I'm saying what you reported looks like a straight up bug, which may have been fixed by now in dev.

Basically, the error you're seeing is of course very unclear, but I don't think there should have been such an error in the first place.

**Comment by nlarusstone** [comment]
It was a real error that the linter was catching--I had an unmatched parenthesis. That's why I don't think it's a bug, but rather a messaging improvement.

**Comment by MichaelChirico** [comment]
That makes sense -- could you please provide a reproducible example to see
how to proceed?

**Comment by MichaelChirico** [comment]
>1 year with no reprex -- feel free to re-open if you've got an example that'll make it easier to debug. Closing till then.
